ethr/ethr.go

```
@@ -0,0 +1,473 @@
+				match := attributeMatcher.FindStringSubmatch(attributeName)
+				if len(match) > 0 {
+					section := match[1]
+					algorithm := match[2]
+					kind := match[4]
+					legacy, ok := legacyAttrTypes[match[4]]
+					if ok {
+						kind = legacy
+					}
+					encoding := match[6]
+					keyOrService := string(event.Value[:])
+					if strings.HasPrefix(keyOrService, "0x") {
+						keyOrService = keyOrService[2:]
+					}
+					switch section {
+					case "pub":
+						delegateCount++
+						pk := resolver.VerificationMethod{
+							ID:         fmt.Sprintf("%s#delegate-%d", did, delegateCount),
+							Controller: did,
+							Type:       algorithm,
+						}
+						legacy, ok := legacyAlgoMap[algorithm+kind]
+						if ok {
+							pk.Type = legacy
+						}
+						switch encoding {
+						case "", "hex":
+							keyOrServiceBytes, err := hex.DecodeString(keyOrService)
+							if err != nil {
+								return nil, err
+							}
+							publicKeyMultibase, err := multibase.Encode(multibase.Base16, keyOrServiceBytes)
+							if err != nil {
+								return nil, err
+							}
+							pk.PublicKeyMultibase = publicKeyMultibase
+						case "base64":
+							keyOrServiceBytes, err := hex.DecodeString(keyOrService)
+							if err != nil {
+								return nil, err
+							}
+							publicKeyMultibase, err := multibase.Encode(multibase.Base64, keyOrServiceBytes)
+							if err != nil {
+								return nil, err
+							}
+							pk.PublicKeyMultibase = publicKeyMultibase
+						case "base58":
+							keyOrServiceBytes, err := hex.DecodeString(keyOrService)
+							if err != nil {
+								return nil, err
+							}
+							publicKeyMultibase, err := multibase.Encode(multibase.Base58BTC, keyOrServiceBytes)
+							if err != nil {
+								return nil, err
+							}
+							pk.PublicKeyMultibase = publicKeyMultibase
+						case "pem":
+							pk.PublicKeyPem = keyOrService
+						default:
+							// TODO: Should we have a default case?
```

Should we return an error we can't understand the key encoding?